tensorflow_datasets/core/custom_dataset/image_folder_test.py

```diff
@@ -20,30 +20,36 @@
 from __future__ import division
 from __future__ import print_function
 
-import functools
-from tensorflow_datasets import testing
-from tensorflow_datasets.core import registered
-from tensorflow_datasets.image_classification import image_folder
+import os
 
+import tensorflow_datasets.public_api as tfds
 
-class ImageLabelFolderTest(testing.DatasetBuilderTestCase):
+original_init = tfds.ImageLabelFolder.__init__
+
+def new_init(self, root_dir=None, **kwargs):
+  assert root_dir is None
+  del kwargs
+  root_dir = os.path.join(
+      tfds.core.utils.tfds_dir(), 'testing',
+      'test_data', 'fake_examples', 'image_label_folder')
+  original_init(self, root_dir=root_dir)
+
+tfds.ImageLabelFolder.__init__ = new_init
+
+class ImageLabelFolderTest(tfds.testing.DatasetBuilderTestCase):
   """Test for ImageLabelFolder."""
-  DATASET_CLASS = functools.partial(
-      image_folder.ImageLabelFolder, dataset_name="image_folder_data")
-  # The above construct forces us to disable those checks:
-  MOCK_OUT_FORBIDDEN_OS_FUNCTIONS = False
+
+  DATASET_CLASS = tfds.ImageLabelFolder
+
   SPLITS = {
       "train": 2,  # Number of examples.
       "test": 6,
   }
 
-  def test_info(self):
-    pass
-
   def test_registered(self):
-    self.assertIn("image_label_folder", registered.list_builders(),
-                  "Dataset was not registered.")
+    self.assertNotIn("image_label_folder", tfds.list_builders(),
```

Comment:
I would move this test in a separate `tfds.testing.TestCase` class to separate `DatasetBuilderTestCase ` from `TestCase`

Reply:
I was overriding a test in `DatasetBuilderTestCase`.
So, should I disable this test and add it in a separate class?